apps/entity-service/src/entities/processors/site.processor.ts

```diff
@@ -1,24 +1,161 @@
-import { EntityProcessor, PaginatedResult } from "./entity-processor";
-import { Site } from "@terramatch-microservices/database/entities";
-import { SiteFullDto, SiteLightDto } from "../dto/site.dto";
+import { DocumentBuilder } from "@terramatch-microservices/common/util";
+import { Aggregate, aggregateColumns, EntityProcessor, PaginatedResult } from "./entity-processor";
+import {
+  Demographic,
+  DemographicEntry,
+  Media,
+  ProjectUser,
+  Seeding,
+  Site,
+  SitePolygon,
+  SiteReport,
+  TreeSpecies
+} from "@terramatch-microservices/database/entities";
+import { AdditionalSiteFullProps, SiteFullDto, SiteLightDto, SiteMedia } from "../dto/site.dto";
+import { EntityQueryDto } from "../dto/entity-query.dto";
+import { BadRequestException } from "@nestjs/common";
+import { FrameworkKey } from "@terramatch-microservices/database/constants/framework";
+import { Op } from "sequelize";
+import { sumBy } from "lodash";
 
 export class SiteProcessor extends EntityProcessor<Site, SiteLightDto, SiteFullDto> {
   readonly LIGHT_DTO = SiteLightDto;
   readonly FULL_DTO = SiteFullDto;
 
-  async findOne(): Promise<Site> {
-    throw new Error("Method not implemented.");
+  async findOne(uuid: string) {
+    return await Site.findOne({
+      where: { uuid },
+      include: [
+        { association: "framework" },
+        { association: "project", attributes: ["name"], include: [{ association: "organisation" }] }
+      ]
+    });
   }
 
-  async findMany(): Promise<PaginatedResult<Site>> {
-    throw new Error("Method not implemented.");
+  async findMany(query: EntityQueryDto, userId?: number, permissions?: string[]): Promise<PaginatedResult<Site>> {
+    const builder = await this.entitiesService.buildQuery(Site, query, [
+      { association: "project", attributes: ["name"] },
+      { association: "framework" }
+    ]);
+
+    if (query.sort != null) {
+      if (["name"].includes(query.sort.field)) {
```

Comment:
If this only needs to compare against one value, please `===` instead of `array.includes`